pkg/controller/constrainttemplate/constrainttemplate_controller.go

```diff
@@ -426,6 +480,87 @@ func (r *ReconcileConstraintTemplate) handleDelete(
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileConstraintTemplate) defaultGetPod() (*corev1.Pod, error) {
+	ns := util.GetNamespace()
+	name := util.GetPodName()
+	key := types.NamespacedName{Namespace: ns, Name: name}
+	pod := &corev1.Pod{}
+	if err := r.Get(context.TODO(), key, pod); err != nil {
+		return nil, err
+	}
+	return pod, nil
+}
+
+func (r *ReconcileConstraintTemplate) deleteAllStatus(ctName string) error {
+	statusObj := &statusv1beta1.ConstraintTemplatePodStatus{}
+	statusObj.SetName(statusv1beta1.KeyForConstraintTemplate(util.GetPodName(), ctName))
+	statusObj.SetNamespace(util.GetNamespace())
+	if err := r.Delete(context.TODO(), statusObj); err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+	}
+
+	cstrStatusObjs := &statusv1beta1.ConstraintPodStatusList{}
```

Comment:
It looks like both the constraint controller and the constrainttemplate controllers are responsible for deleting the same status object?

Reply:
Correct, because the constraint controller cannot be trusted to delete a status object once its watch has been deregistered. I figure repetitive deletes are fine, as it should be an idempotent operation anyway.

Reply:
I should not be returning nil on error though... oops